users/api_views.py

```diff
@@ -7,3 +18,26 @@
 class UserAPIRegistration(CreateAPIView):
     queryset = User.objects.all()
     serializer_class = UserSerializer
+
+
+class UserAPILogin(TokenObtainPairView):
+    def post(self: "UserAPILogin", request: Request, *args: tuple, **kwargs: dict[str, Any]) -> Response:
+        response = super().post(request, *args, **kwargs)
+        user = authenticate(username=self.request.data["email"], password=self.request.data["password"])
+        login(request, user)
+        return response
+
+
+class UserAPILogout(TokenRefreshView):
+    authentication_classes: ClassVar[list[str]] = [SessionAuthentication]
```

Comment:
Wow wait, please explain. Session authentication when we are logging out users by JWT token?
Session auth is cookie based, in API realm, there are no cookies. Session auth is used for "classic" django views, where we use HTTP sites, not for API based views. Or am I not seeing something?

Here is some documentation: https://github.com/jazzband/djangorestframework-simplejwt/blob/9911c85e13eda3f98396923d94342c6f9e5e8d8b/docs/stateless_user_authentication.rst#L24.
JWT auth is stateless, SessionAuth is not stateless. Those are two different authentication methods.

I am thinking we are missing this setting: https://github.com/jazzband/djangorestframework-simplejwt/blob/9911c85e13eda3f98396923d94342c6f9e5e8d8b/docs/getting_started.rst#project-configuration